src/Sarif.Viewer.VisualStudio/CodeFlowToTreeConverter.cs

```diff
@@ -4,14 +4,57 @@
 using System;
 using Microsoft.CodeAnalysis.Sarif;
 using Microsoft.Sarif.Viewer.Models;
+using System.Collections;
+using System.Collections.Generic;
 
 namespace Microsoft.Sarif.Viewer.VisualStudio
 {
     internal static class CodeFlowToTreeConverter
     {
         internal static CallTreeNode Convert(CodeFlow codeFlow)
         {
-            throw new NotImplementedException();
+           int i = -1;
+
+            CallTreeNode root = new CallTreeNode
+            {
+                Children = GetChildren(codeFlow, ref i)
+            };
+
+            return root;
+        }
+
+        private static List<CallTreeNode> GetChildren(CodeFlow codeFlow, ref int i)
+        {
+            i++;
+            List<CallTreeNode> children = new List<CallTreeNode>();
+            while (i < codeFlow.Locations.Count )
```

Comment:
A couple of suggestions on structuring:

1. If you use a `switch`, you won't have to repeat the expression `codeFlow.Locations[i].Kind` in each branch.
2. I'd expect to see calls before returns -- obviously it doesn't matter, but it reads more naturally to me that way -- so I'd handle the `Call` case first.

The only problem is that this switch is enclosed within a `while` loop, and you need to break out of the `while` loop when you find a `CallReturn`. And since C# doesn't have a "multi-level break" statement, you'll need an auxilliary flag.

So, something like this:

```C#
bool foundCallReturn = false;
while (i < codeFlowLocations.Count && !foundCallReturn)
{
    switch (codeFlow.Locations[i].Kind)
    {
        case AnnotatedCodeLocationKind.Call:
            ....
            break;

        case AnnotatedCallLocation.CallReturn:
            ...
            foundCallReturn = true;
            break;

        default:
            ...
            break;
    }
}
```